src/Stryker.Core/Stryker.Core/MutationTest/MutationTestExecutor.cs

```diff
@@ -13,45 +13,97 @@ namespace Stryker.Core.MutationTest
     public interface IMutationTestExecutor
     {
         ITestRunner TestRunner { get; }
-        void Test(Mutant mutant, int timeoutMs);
+        void Test(IList<Mutant> mutant, int timeoutMs, TestUpdateHandler updateHandler);
     }
 
     public class MutationTestExecutor : IMutationTestExecutor
     {
         public ITestRunner TestRunner { get; }
         private ILogger Logger { get; }
 
-        public IEnumerable<TestDescription> Tests => TestRunner?.Tests;
-
         public MutationTestExecutor(ITestRunner testRunner)
         {
             TestRunner = testRunner;
             Logger = ApplicationLogging.LoggerFactory.CreateLogger<MutationTestProcess>();
         }
 
-        public void Test(Mutant mutant, int timeoutMs)
+        public void Test(IList<Mutant> mutantsToTest, int timeoutMs, TestUpdateHandler updateHandler)
         {
-            try
+            var forceSingle = false;
+            while(mutantsToTest.Any())
             {
-                Logger.LogDebug($"Testing {mutant.LongName}.");
-                var result = TestRunner.RunAll(timeoutMs, mutant);
-                Logger.LogTrace($"Testrun for {mutant.DisplayName} with output {result.ResultMessage}");
+                var result = RunTestSession(mutantsToTest, timeoutMs, updateHandler, forceSingle);
+
+                Logger.LogDebug(
+                    $"Test run for {string.Join(" ,", mutantsToTest.Select(x => x.DisplayName))} is {(result.FailingTests.Count == 0 ? "success" : "failed")} with output: {result.ResultMessage}");
 
-                mutant.ResultStatus = result.Success ? MutantStatus.Survived : MutantStatus.Killed;
+                var remainingMutants = mutantsToTest.Where( (m) => m.ResultStatus == MutantStatus.NotRun).ToList();
+                if (remainingMutants.Count == mutantsToTest.Count)
+                {
+                    // the test failed to get any conclusive results
+                    if (!result.SessionTimedOut)
+                    {
+                        // something bad happened.
+                        Logger.LogError($"Stryker failed to test {remainingMutants.Count} mutant(s).");
+                        return;
+                    }
+
+                    // test session's results have been corrupted by the time out
+                    // we retry and run tests one by one, if necessary
+                    if (remainingMutants.Count == 1)
```

Comment:
actually that's the logic: we removed every mutation for which the status has been determined.
plus, if there was a test active when the timeout triggers, we mark the associated mutation as `TimeOut` 
But my testing revealed that sometimes we get a timeout without having any results or any active test.
Probably because the endless loop somehow prevented vstest to report progress.

When this happen, here the logic is to fallback to a cautious approach by running each mutation on its own. I chosen this strategy to avoid having multiple timeouts by re running the same configuration. This is quite rare and random.